adjacency_list.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'edge_list'
+require_relative 'linkedlist'
+
+class AdjacencyList
+  attr_reader :list, :first
+  def initialize(list)
+    @list = []
+    build_linked_list(list)
+  end
+
+  def build_linked_list(list)
+    list.each do |row|
```

Comment:
you can do it in one step:
`list.each do |from, to, weight|`
and remove the next 3 statements.